src/execution_plan/execution_plan.c

```diff
@@ -422,16 +420,21 @@ ExecutionPlan* _NewExecutionPlan(RedisModuleCtx *ctx, AST *ast, ResultSet *resul
                 AlgebraicExpression **exps = AlgebraicExpression_From_QueryGraph(cc, ast, &expCount);
 
                 // Reorder exps, to the most performent arrangement of evaluation.
-                orderExpressions(exps, expCount, filter_tree);
+                orderExpressions(exps, expCount, execution_plan->filter_tree);
 
                 AlgebraicExpression *exp = exps[0];
-                selectEntryPoint(exp, filter_tree);
+                selectEntryPoint(exp, execution_plan->filter_tree);
 
                 // Create SCAN operation.
                 if(exp->src_node->label) {
-                    /* There's no longer need for the last matrix operand
-                     * as it's been replaced by label scan. */
-                    AlgebraicExpression_RemoveTerm(exp, 0, NULL);
+                    /* Resolve source node by performing label scan,
+                     * in which case if the first algebraic expression operand
+                     * is a label matrix (diagonal) remove it, otherwise
+                     * the label matrix associated with source's label is located
+                     * within another traversal operation, for the timebeing do not
+                     * try to locate and remove it, there's no real harm except some performace hit
+                     * in keeping that label matrix. */
```

Comment:
In the future this should be pretty easy to resolve, I think. For now it's no big deal, as you say!